docs/documentation/dependent-resources.md

```diff
@@ -0,0 +1,115 @@
+---
+title: Dependent Resources Feature description: Dependent Resources Feature layout: docs permalink:
+/docs/dependent-resources
+---
+
+# Dependent Resources
+
+DISCLAIMER: The Dependent Resource support is relatively new and, while we strove to cover what we
+anticipate will be the most common use cases, the implementation is not simple and might still
+evolve. As a result, some APIs could be still a subject of change in the future. However,
+non-backwards compatible changes are expected to be trivial to adapt to.
+
+## Motivations and Goals
+
+Most operators need to deal with secondary resources when trying to realize the desired state
+described by the primary resource it is in charge of. For example, the Kubernetes-native
+`Deployment` controller needs to manage `ReplicaSet` instances as part of a `Deployment`'s
+reconciliation process. In this instance, `ReplicatSet` is considered a secondary resource for
+the `Deployment` controller.
+
+Controllers that deal with secondary resources typically need to perform the following steps, for
+each secondary resource:
+
+```mermaid
```

Comment:
I'm pretty sure Mermaid will not work with Jekyll

Reply:
Yes but we should be able to make it work by importing the JS lib. See https://codewithhugo.com/mermaid-js-hugo-shortcode/ for example.

Reply:
Great, that would be awesome.